src/Stratis.Bitcoin.Features.SmartContracts.Tests/ContractUpgradeabilityTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Emit;
+using Xunit;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.Tests
+{
+    public class ContractUpgradeabilityTests
+    {
+        public class TestAlc : AssemblyLoadContext
+        {
+            protected override Assembly Load(AssemblyName assemblyName)
+            {
+                return null;
+            }
+        }
+
+        [Fact]
+        public void Load_Contract_Compiled_Against_V1_With_V2_SmartContracts()
```

Comment:
extend test name here to explain what happens in this scenario,  eg  it fails, it works fine ?